lms/djangoapps/ccx/overrides.py

```diff
@@ -142,18 +142,15 @@ def override_field_for_ccx(ccx, block, name, value):
         override_has_changes = serialized_value != override.value
 
     if not override:
-        try:
-            override = CcxFieldOverride.objects.create(
-                ccx=ccx,
-                location=block.location,
-                field=name,
-                value=serialized_value
-            )
+        override, created = CcxFieldOverride.objects.get_or_create(
+            ccx=ccx,
+            location=block.location,
+            field=name,
+            defaults = {'value': serialized_value},
+        )
+        if created:
             _get_overrides_for_ccx(ccx).setdefault(block.location, {})[name + "_id"] = override.id
-        except IntegrityError:
-            transaction.commit()
-            kwargs = {'ccx': ccx, 'location': block.location, 'field': name}
-            override = CcxFieldOverride.objects.get(**kwargs)
+        else:
```

Comment:
Instead of trying to `create` an object first and do a `get` on an `IntegrityError`, lets just do a `get` first.